server/health.go

```diff
@@ -56,15 +56,24 @@ func (s *healthService) Check(ctx context.Context, in *healthpb.HealthCheckReque
 		return &healthpb.HealthCheckResponse{Status: healthpb.HealthCheckResponse_UNKNOWN}, status.Errorf(codes.Internal, "failed to get clone status: err=%v", err)
 	}
 
-	var isOutOfSynced, isUnderCloning bool
-	if replicaStatus != nil && replicaStatus.LastIoErrno != 0 {
+	var isOutOfSynced, hasSQLThreadError, isUnderCloning bool
+	if replicaStatus != nil &&
+		replicaStatus.SlaveIOState != "" &&
+		(replicaStatus.LastIoErrno != 0 || replicaStatus.SlaveIORunning != moco.ReplicaRunConnect) {
 		isOutOfSynced = true
 	}
+
+	if replicaStatus != nil &&
+		replicaStatus.SlaveIOState != "" &&
```

Comment:
`Replica_IO_State` (=SlaveIOState) shows the state of the replication process.  When `SLAVE STOP`ed and NOT restarting replication, this column becomes the empty string, it can be utilized whether the instance works in primary or replica mode (`Replica_SQL_State` column doesn't exist).
https://dev.mysql.com/doc/refman/8.0/en/replica-io-thread-states.html

We should check the `Last_{IO|SQL}_Errno` only if the instance works in replica mode.

Reply:
Thank you. I had a misunderstanding.
In my opinion, it is more understandable to write the following.

```
        // comment
	var hasIOThreadError, hasSQLThreadError bool
	if replicaStatus != nil && replicaStatus.SlaveIOState != "" {
		if replicaStatus.LastIoErrno != 0 || replicaStatus.SlaveIORunning != moco.ReplicaRunConnect {
			hasIOThreadError = true
		}
		if replicaStatus.LastSQLErrno != 0 || replicaStatus.SlaveSQLRunning != moco.ReplicaRunConnect {
			hasSQLThreadError = true
		}
	}
	
	var isUnderCloning bool
	if ... {
```